Public/Src/Engine/Processes/Internal/Pipes.cs

```diff
@@ -108,17 +108,48 @@ public static void CreateInheritablePipe(PipeInheritance inheritance, PipeFlags
                 readSideFlags |= FileFlagsAndAttributes.FileFlagOverlapped;
             }
 
-            OpenFileResult openReadSideResult = FileUtilities.TryCreateOrOpenFile(
-                pipeName,
-                FileDesiredAccess.GenericRead,
-                FileShare.None,
-                FileMode.Open,
-                readSideFlags,
-                out readHandle);
+            int maxRetry = 3;
 
-            if (!openReadSideResult.Succeeded)
+            while (true)

```

Comment:
don't we have some generic utility method for retrying an arbitrary action?

Reply:
maybe not, or at least I couldn't find anything

Reply:
Actually `RetryOnFailure` from the IO helpers could be used here?

Reply:
The retry logic here is a bit different from `RetryOnFailure`. We are doing wait for at most 5 sec in a loop, and then retry.